server/src/test/java/org/elasticsearch/index/query/MatchPhrasePrefixQueryBuilderTests.java

```diff
@@ -85,6 +91,12 @@ protected MatchPhrasePrefixQueryBuilder doCreateTestQueryBuilder() {
     protected void doAssertLuceneQuery(MatchPhrasePrefixQueryBuilder queryBuilder, Query query, QueryShardContext context)
             throws IOException {
         assertThat(query, notNullValue());
+
+        if (query instanceof MatchAllDocsQuery) {
```

Comment:
As we are asserting Lucene queries there, I would make this test in reverse:

if (queryBuilder.zeroTermsQuery().equals(ZeroTermsQuery.ALL)) {
    assert that query instanceof MatchAllDocsQuery;
}

Reply:
I can understand the ask, but unfortunately there are several other conditions the random queryBuilder needs to fullfil to actually produce a MatchAllDocsQuery. Its quite difficult to get them all right, wheras the test the other way round is straight forward. Would you object to leaving it as is?

Reply:
An alternative would be not to test this here at all since we have another dedicated test for the `zeroTermsQuery` behaviour. In that case I'd just add `MatchAllDocsQuery` to the set of allowed and expected queries here.

Reply:
@cbuescher Thanks for the explanation, I am ok to leave it as is.